### PR TITLE
[FLINK-23529] Add Flink 1.13 MigrationVersion

### PR DESCRIPTION
**Pull request by tsreaper**
## What is the purpose of the change

Currently the largest `MigrationVersion` is 1.12. We need newer versions to add more serializer compatibility tests.

This PR adds 1.13 `MigrationVersion`.

## Brief change log

- Add 1.13 MigrationVersion

## Verifying this change

This change added tests and can be verified by running the added tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/AbstractOperatorRestoreTestBase.java [hunk 1]
```diff
@@ -177,6 +177,9 @@ private String migrateJob(ClusterClient<?> clusterClient, Deadline deadline) thr
                 JobStatus.CANCELED,
                 jobCanceledFuture.get(deadline.timeLeft().toMillis(), TimeUnit.MILLISECONDS));
 
+        // TODO: the savepoint _metadata file for the current branch is stored in this
+        //  savepointPath, please create the corresponding test resource directory and
+        //  copy the _metadata file by hand.
```
Comment (xintongsong) [inline flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/AbstractOperatorRestoreTestBase.java:182]: I'd suggest to make this change a separate hotfix commit. Additionally, we probably should move this to the JavaDoc of the class.

**Review by xintongsong — COMMENTED** [review]
Thanks for the PR, @tsreaper. The changes LGTM in general. I have only one minor comment. Please take a look.

**Comment by tsreaper** [comment]
@flinkbot run azure

**Comment by xintongsong** [comment]
@flinkbot run azure

**Comment by xintongsong** [comment]
@flinkbot run azure

**Comment by tsreaper** [comment]
> @flinkbot run azure

It already succeeded in https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=21395&view=results

**Comment by xintongsong** [comment]
Thanks, @tsreaper. Merging this.

**Pull request closed without merging**